File: frontend_web/app/types/duel.ts
export interface PlayerData {
  playerId: string;
  elo: number;
  rank: string;
}

export interface RequestDuelResponse {
  duelId: string;
  message: string;
}

export interface AcceptDuelResponse {
  duelId: string;
  message: string;
}

export interface Category {
  name: string;
  displayName: string;
  description: string;
}

export interface RequestDuelInput {
  requesterId: string;
  opponentId: string;
  category: string;
}

export interface AcceptDuelInput {
  duelId: string;
}

File: api_gateway/app/graphql/types/duel.py
"""
GraphQL types for the Duel Service
"""

import strawberry
from typing import Optional


@strawberry.type
class PlayerData:
    playerId: str = strawberry.field(name="playerId")
    elo: int
    rank: str


@strawberry.type
class RequestDuelResponse:
    duelId: str = strawberry.field(name="duelId")
    message: str


@strawberry.type
class AcceptDuelResponse:
    duelId: str = strawberry.field(name="duelId")
    message: str


@strawberry.type
class Category:
    name: str
    displayName: str = strawberry.field(name="displayName")
    description: str


@strawberry.type
class ErrorResponse:
    error: str


@strawberry.input
class RequestDuelInput:
    requesterId: str = strawberry.field(name="requesterId")
    opponentId: str = strawberry.field(name="opponentId")
    category: str


@strawberry.input
class AcceptDuelInput:
    duelId: str = strawberry.field(name="duelId")
